src/otx/core/config/data.py

```diff
@@ -2,10 +2,10 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 """Config data type objects for data."""
-from __future__ import annotations
-
+# NOTE: omegaconf would fail to parse dataclass with `from __future__ import annotations` in Python 3.8, 3.9
+# ruff: noqa: FA100
```

Comment:
@harimkang @vinnamkim

Python 3.10 CI test runs fine with `from __future__ import annotations`. However, for Python 3.8 and 3.9, the usage of this import statement triggers an error, potentially related to PEP 604. To address this, I have temporarily modified the file to employ Python 3.8 type-checking conventions.

Please take a look at the provided error traceback, and if you have a more optimal solution or insights into the compatibility issue, I'd be greatly appreciated.
```
  File "/home/yuchunli/git/otx-2x/venv_39/lib/python3.8/site-packages/hydra/core/config_store.py", line 85, in store
    cfg = OmegaConf.structured(node)
  File "/home/yuchunli/git/otx-2x/src/otx/core/config/__init__.py", line 118, in register_configs
    cs.store(group="data", name="base_inst_seg", node=InstSegDataModuleConfig)
  ...
  omegaconf.errors.ConfigTypeError: 'type' object is not subscriptable
    full_key: 
    object_type=None
```